java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/CompositeJdbcConsumer.java

```diff
@@ -43,7 +46,18 @@ public CompositeJdbcConsumer(JdbcConsumer[] consumers) {
   @Override
   public void consume(ResultSet rs) throws SQLException, IOException {
     for (int i = 0; i < consumers.length; i++) {
-      consumers[i].consume(rs);
+      try {
+        consumers[i].consume(rs);
```

Comment:
just curious what if `consume` method throws the defined exception rather than catching and throwing it back?

Reply:
What do you mean?

If you mean that every consumer that extends BaseConsumer should implement its own `try/catch` to throw the `JdbcConsumerException` within its `consume` method then I think it's doable, but it would require more changes and require more effort when implementing a custom consumer (users would need to ensure they add the try/catch logic as well). Handling it here seems easier to both implement and use since custom consumers can just extend `BaseConsumer`.

Reply:
I understand your point. Let's keep it as it is.